en/docs/learn/identifier-first-flow-handler.md

```diff
@@ -0,0 +1,91 @@
+
+        ```bash
+        127.0.0.1 localhost.com
+        ```
+    5.  Restart WSO2 Identity Server.
+
+1.  Download the saml2-web-app-pickup-dispatch.com.war file from [here](https://github.com/wso2/samples-is/releases) and copy it inside the <TOMCAT_HOME>/webapps directory.
+
+2.  Start the tomcat server.  
+
+3.  In the WSO2 IS [management console](https://localhost:9443/carbon/admin/login.jsp), create a new service provider and expand the Inbound Authentication configuration. Select the SAML2 Web SSO configuration section, and click Configure.
+
+4.  Provide the following details and register.
+    -   Issuer- \saml2-web-app-pickup-dispatch.com
+    -   Assertion Consumer URL- \http://localhost.com:8080/saml2-web-app-pickup-dispatch.com/consumer
+
+5.  Access the PickUp application URL at http://localhost.com:8080/saml2-web-app-pickup-dispatch.com. The login screen should only prompt for the username. Only after entering the correct username it proceeds to the next screen where you are requested to enter the password. 
+    ![only-username](../assets/img/learn/only-username.png)
+
+6.  Now let's try to incorporate a federated authenticator in this scenario. In the management console, configure the federated authenticators required. Details on how to configure these can be found [here](../../learn/configuring-federated-authentication/).  
+
+7.  Open the Service Provider you created in step 3 and proceed to the **Advanced Configuration**. Expand  **Local & Outbound Authentication Configuration**, add identifier-first as the first step and, add all the federated authenticators as another step.  
+
+![identifier-federation](../assets/img/learn/identifier-first-fed.png)
+
+8.  Add the following script to **Script Based Adaptive Authentication** in the **Advanced Configuration** of the Service Provider. This extracts the domain name from the user name in the first step and uses that particular domain as the authenticator in the second step. 
+
+```java 
+var federatedDomains = ['facebook.com', 'yahoo.com', 'gmail.com'];
+function onLoginRequest(context) {
+ executeStep(1, {
+    onSuccess: function (context) {
+       var username = context.steps[1].subject.username;
+       var indexOfLastAt = username.lastIndexOf("@");
+       var domain = username.substring(indexOfLastAt + 1);
+       if (federatedDomains.indexOf(domain) >= 0) {
+          executeStep(2,{authenticationOptions:[{idp:domain}]}, {});
+       } else {
+          executeStep(2,{authenticationOptions : [{authenticator:'basic'}]} , {});
+       }
+    }
+ });
+}
+```
+
+Now you can access the sample app using any of the configured federated authenticators, by entering the username followed by the authenticator (for example, jake@gmail.com). You can either disable the `validate_username` property and test it with any name or you can create a user in the console. See more details on how to use an email address as the username [here](../../learn/using-email-address-as-the-username).
```

Comment:
Use either "username" or "user name" consistently throughout the document. It has been spelled as one word in some instances and as two words in another instance.